src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1474,6 +1476,47 @@ protected override void InvokeDefaultAction(string path)
             return true;
         }
 
+        //Returns true if a given DirectoryInfo path is a recursive sym link.
+        private bool IsRecursiveSymLink(DirectoryInfo path)
+        {
+            string readSymLink = "";
+
+            if (Platform.IsWindows)
+            {
+                PSObject obj = PSObject.AsPSObject(path);
+
+                try
+                {
+                    readSymLink = InternalSymbolicLinkLinkCodeMethods.GetTarget(obj).First().ToString();
+                }
+                catch
+                {
+                    readSymLink = null;
+                }
+
+                //if the read symlink is not null or empty, and what it points to is the parent directory then it is recursive
+                if (!String.IsNullOrEmpty(readSymLink) && path.Parent.FullName.Contains(readSymLink))
```

Comment:
And also, please use the `StartsWith` overload with the culture comparison parameter.